pkg/instrument/extension.h

```diff
@@ -0,0 +1,41 @@
+#include <sqlite3.h>
+#include <stdint.h>
+
+typedef int sqlite3_error;
+
+typedef struct sqlite3_metrics_s {
+    uint64_t pages_cache_write;
+    uint64_t pages_cache_hit;
+    uint64_t pages_cache_miss;
+    uint64_t pages_cache_spill;
+
+    uint64_t read_txn_time_ns;
+    uint64_t write_txn_time_ns;
+} sqlite3_metrics_t;
+
+// sqlite3_instrument register an auto extension that 
+// instruments all new connections to collect key sqlite
+// performance metrics. Existing connections will not be 
+// affected. This call uses the trace hook to collect data.
+// Replacing the hook will remove instrumentation for the
+// connection. In this case, it is possible to call 
+// [sqlite3_collect_metrics] to keep the instrumentation
+// active in a custom trace hook.
+sqlite3_error sqlite3_instrument();
```

Comment:
`s/register/registers/`
`s/an auto extension that/an auto extension which/`

What does 'key' sqlite performance metrics mean here? Is it key-performance metrics or 'key' (as in 'important') sqlite metrics. If it's the latter, I think 'key' is very unspecific so should be dropped.

There's some slight inconsistency here, the doc comments for `sqlite3_instrument` and `_uninstrument` mention different hooks, could these be made the same? Also, do we need to mention which hooks are set? Could it be kept as an implementation detail, or could this cause unexpected clashes?

Reply:
We need to keep the description of the hooks as only one hook of each kind can be set up (i.e. if someone does, he will remove instrumentation from that connection).

Regarding the 'key` it's as in 'important metrics'. What do you suggest instead? Just dropping `key`?

Reply:
I think dropping the word 'key' would be cleanest